src/hipscat/pixel_tree/pixel_tree.py

```diff
@@ -57,125 +56,56 @@ def contains(self, pixel: HealpixInputTypes) -> bool:
         Returns:
             True if the tree contains the pixel, False if not
         """
-        pixel = get_healpix_pixel(pixel)
-        return pixel.order in self.pixels and pixel.pixel in self.pixels[pixel.order]
+        (order, pixel) = get_healpix_tuple(pixel)
+        if order > self.tree_order:
+            return False
+        d_order = self.tree_order - order
+        pixel_at_tree_order = pixel << 2 * d_order
+        index = np.searchsorted(self.tree.T[1], pixel_at_tree_order, side="right")
+        if index >= len(self.pixels):
+            return False
+        is_same_order = self.pixels[index][0] == order
+        return pixel_at_tree_order == self.tree[index][0] and is_same_order
```

Comment:
Yep, that's right. That's how `contains` was implemented before and I didn't want to change it. I think it makes sense, but it might be nice to have a function that checks if there is overlapping too.